Cabal/src/Distribution/Simple/GHC.hs

```diff
@@ -246,10 +248,16 @@ configure verbosity hcPath hcPkgPath conf0 = do
 
       filterExt ext = filter ((/= EnableExtension ext) . fst)
 
+      compilerId :: CompilerId
+      compilerId = CompilerId GHC ghcVersion
+
+      compilerAbiTag :: AbiTag
+      compilerAbiTag = maybe NoAbiTag AbiTag (Map.lookup "Project Unit Id" ghcInfoMap >>= stripPrefix (prettyShow compilerId <> "-"))
```

Comment:
@bgamari is "Project Unit Id" the proper ABI identifier?

I see `ghc-9.8.1-f7d8` in mine.

Reply:
Ah, I get ```
❯ ghc-9.8.1 --info | grep Unit
 ,("Project Unit Id","ghc-9.8.1-5640")
```

Reply:
Let me understand. GHC does not actually offer anything called ABI tag/identifier/hash? We are going to use this `Project Unit Id`? Is this the right thing to do?

Reply:
`Project Unit Id` is indeed a reasonable thing to use here. Indeed it was specifically added to avoid mixing compilation artifacts from different compilers.